mm2src/coins/lightning/ln_events.rs

```diff
@@ -219,7 +220,7 @@ fn sign_funding_transaction(
         .activated_key_or_err()
         .map_err(|e| SignFundingTransactionError::Internal(e.to_string()))?;
 
-    let prev_script = Builder::build_p2pkh(&my_address.hash);
+    let prev_script = output_script(my_address);
```

Comment:
I suppose this change will break sighash calculation, as `output_script` will return `build_witness_script` for Segwit address. As per https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#native-p2wpkh, the scriptCode is still `OP_DUP OP_HASH160 hash160(pubkey) OP_EQUALVERIFY OP_CHECKSIG` for Native P2WPKH input signing.

There are also test failures related to similar changes in other places: https://github.com/KomodoPlatform/komodo-defi-framework/actions/runs/6096406569/job/16541897281?pr=1960#step:5:1529

Reply:
Ah, sorry, I fixed that segwit signing code for this case in my original PR but did not make it here.
(I ran tests but apparently there are more tests in nested crates that I missed)